src/solver/pad_constant/pad_constant_fwd_contiguous.cpp

```diff
@@ -0,0 +1,136 @@
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "miopen/conv_solution.hpp"
+#include "miopen/datatype.hpp"
+#include "miopen/execution_context.hpp"
+#include "miopen/kernel_build_params.hpp"
+#include "miopen/miopen.h"
+#include "miopen/mlo_internal.hpp"
+#include "miopen/pad_constant/solvers.hpp"
+#include "miopen/pad_constant/invoke_params.hpp"
+
+namespace miopen {
+namespace solver {
+namespace pad_constant_fwd_contiguous {
+bool PadConstantFwdContiguous::IsApplicable(
+    const ExecutionContext& /*context*/,
+    const miopen::pad_constant_fwd_contiguous::ProblemDescription& problem) const
+{
+    if(!problem.IsSameType())
+    {
+        return false;
+    }
+
+    return true;
+}
+
+ConvSolution PadConstantFwdContiguous::GetSolution(
+    const ExecutionContext& /*context*/,
+    const miopen::pad_constant_fwd_contiguous::ProblemDescription& problem) const
+{
+    auto result = ConvSolution{miopenStatusSuccess};
+    auto ydims  = problem.GetYDesc().GetLengths();
+
+    auto input_dtype  = miopen::GetDataType(problem.GetXDesc().GetType());
+    auto output_dtype = miopen::GetDataType(problem.GetYDesc().GetType());
+
+    // for xgridsize: 5d -> 1d
+    size_t output_size = 1;
+    for(int i = 0; i < 5; i++)
+    {
+        output_size *= ydims[i] == 0 ? 1 : ydims[i];
+    }
+
+    size_t xlocalsize = 1024;
+    // AlignUp blows up, because output_size can be > int_max. Lovely.
+    // size_t xgridsize  = AlignUp(output_size, xlocalsize);
+    size_t xgridsize =
+        (((static_cast<size_t>(output_size) + xlocalsize - 1) / xlocalsize) * xlocalsize);
+    size_t ylocalsize = 1;
+    size_t ygridsize  = 1;
+
+    auto kernel = KernelInfo{};
+
+    kernel.kernel_file = "MIOpenPadConstantFwd.cpp";
+    kernel.kernel_name = "PadConstantFwdContiguous";
+
+    // TODO: Actually understand how to use this properly
+    const auto build_params = KernelBuildParameters{
+        {"INPUT_TYPE", input_dtype == "half" ? "half" : "float"},
+        {"OUTPUT_TYPE", output_dtype == "half" ? "half" : "float"},
+    };
+
+    kernel.comp_options = build_params.GenerateFor(kbp::HIP{});
+
+    kernel.l_wk.push_back(xlocalsize);
+    kernel.l_wk.push_back(ylocalsize);
+
+    kernel.g_wk.push_back(xgridsize);
+    kernel.g_wk.push_back(ygridsize);
+
+    result.construction_params.push_back(kernel);
+
+    result.invoker_factory = [](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle_, const AnyInvokeParams& invoke_params) {
+            decltype(auto) kernel = handle_.Run(kernels.front());
+            decltype(auto) params =
+                invoke_params.CastTo<miopen::pad_constant_fwd_contiguous::InvokeParams>();
+
+            auto xdims = params.xDesc->GetLengths();
+            auto ydims = params.yDesc->GetLengths();
+
+            // Copy x_dims to GPU (needed to get position in kernel later)
+            const size_t* d_xdims;
```

Comment:
When passing a tensor view, you must assume that you are receiving a 5D tensor. Please refer to https://github.com/ROCm/MIOpen/pull/2883. Please note that quite a bit of modification is required to add a tensor view.

Reply:
That is only meant to be a temporary buffer to pass the dimensions to the kernel. As such, I didn't think it would be necessary to explicitly make it something like an array of 5 elements, as everywhere else (which is the kernel in this case) would already be expecting it to point to a 5d tensor view array. If there are less elements (I don't think there are, since the least you can do is `(1,0,0,0,1)`, which ), that change would definitely be necessary. I'll look into that later.

Reply:
Even if a complex implementation is not required, allocating memory using hipMallocManaged is not an appropriate structure. In general, MIOpen uses workspace to clearly inform the user of the size of memory to be allocated to the GPU. The problem can be solved more easily if we pass a static array of size 5 to the kernel.

Reply:
@seungmanhan Please check if the reworked implementation satisfied the requirements.